src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8513,44 +8519,198 @@ protected MethodGroupResolution ResolveExtension(
+                BoundExpression left,
+                string memberName,
+                int arity,
+                ImmutableArray<TypeWithAnnotations> typeArgumentsWithAnnotations,
+                TypeSymbol? returnType,
+                RefKind returnRefKind,
+                LookupResult lookupResult,
+                AnalyzedArguments? analyzedArguments,
+                ref AnalyzedArguments? actualMethodArguments,
+                ref AnalyzedArguments? actualReceiverArguments,
+                ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo,
+                ref MethodGroupResolution firstResult,
+                OverloadResolution.Options options,
+                CallingConventionInfo callingConvention,
+                LookupResult classicExtensionLookupResult,
+                LookupOptions lookupOptions,
+                Binder binder,
+                ExtensionScope scope,
+                BindingDiagnosticBag diagnostics,
+                out MethodGroupResolution result)
+            {
+                Debug.Assert(left.Type is not null);
+                result = default;
+
+                // 1. gather candidates
+                scope.Binder.LookupExtensionMembersInSingleBinder(
+                    lookupResult, left.Type, memberName, arity,
+                    basesBeingResolved: null, lookupOptions, originalBinder: binder, ref useSiteInfo);
+
+                CompoundUseSiteInfo<AssemblySymbol> classicExtensionUseSiteInfo = binder.GetNewCompoundUseSiteInfo(diagnostics);
+                binder.LookupExtensionMethods(classicExtensionLookupResult, scope, memberName, arity, ref classicExtensionUseSiteInfo);
                 diagnostics.Add(expression, classicExtensionUseSiteInfo);
 
                 lookupResult.MergeEqual(classicExtensionLookupResult);
 
                 if (!lookupResult.IsMultiViable)
                 {
-                    diagnostics.Free();
-                    continue;
+                    return false;
                 }
 
-                var members = ArrayBuilder<Symbol>.GetInstance();
-                Symbol? symbol = this.GetSymbolOrMethodOrPropertyGroupStrict(lookupResult, expression, memberName, arity, members, diagnostics, qualifierOpt: null);
-                if (symbol is not null)
+                // 2. resolve methods
+                MethodGroupResolution methodResult = resolveMethods(expression, left, typeArgumentsWithAnnotations, returnType, returnRefKind,
+                    lookupResult, analyzedArguments, ref actualMethodArguments, options, in callingConvention, binder, diagnostics);
+
+                // 3. resolve properties
+                OverloadResolutionResult<PropertySymbol>? propertyResult = resolveProperties(left, lookupResult, binder, ref actualReceiverArguments, ref useSiteInfo);
+
+                // 4. determine member kind
+                if (!methodResult.HasAnyApplicableMethod && propertyResult?.HasAnyApplicableMember != true)
```

Comment:
>HasAnyApplicableMethod [](http://example.com/codeflow?start=34&length=22)

Implementation of this property is somewhat strange. I think that for the purpose of this check we should use ```methodResult.OverloadResolutionResult?.HasAnyApplicableMember```

Reply:
Yes, there's some subtlety. `OverloadResolutionResult` is `null` when we have no `analyzedArguments`.
We could do `if (methodResult.OverloadResolutionResult?.HasAnyApplicableMember == false && ` here, but that's pretty much what `!methodResult.HasAnyApplicableMethod` does (module a check of LookupResultKind.Viable)
I prefer to stick with the check that is in `main` branch.

A relevant test is `ResolveAll_Instance_LocalDeclaration_InnerExtensionMethodVsOuterExtensionProperty`.

Reply:
> Yes, there's some subtlety.` OverloadResolutionResult` is `null` when we have no `analyzedArguments`.

> I prefer to stick with the check that is in `main` branch.

I guess this would be fine with me. However, I would like us to have tests that demonstrate that an ambiguity with a property in a situation when ` OverloadResolutionResult` is `null` is desired. I do not think ```ResolveAll_Instance_LocalDeclaration_InnerExtensionMethodVsOuterExtensionProperty``` falls into this category. We can follow up offline in more details.